packages/ui/src/ui/hooks/use-before-unload.ts

```diff
@@ -0,0 +1,18 @@
+import {useEffect} from 'react';
+
+const beforeUnloadHandler = (e: BeforeUnloadEvent) => {
+    e.preventDefault();
+
+    // Included for legacy support, e.g. Chrome/Edge < 119
+    e.returnValue = true;
+};
+
+export const useBeforeUnload = ({shouldListen}: {shouldListen: boolean}) => {
```

Comment:
Probably it is better to name the hook `usePreventUnload`

Reply:
renamed it